assert/assertions.go

```diff
@@ -1782,18 +1787,89 @@ func diff(expected interface{}, actual interface{}) string {
 		e = spewConfig.Sdump(expected)
 		a = spewConfig.Sdump(actual)
 	}
+	structuredDiff := structuredDiff(e, a)
+	prettyDiff := prettyDiff(structuredDiff, options.ColoredOutput)
+	return "\n\nDiff:\n" + prettyDiff
+}
+
+func structuredDiff(e string, a string) []diffmatchpatch.Diff {
+	dmp := diffmatchpatch.New()
+	fromRunes, toRunes, runesToLines := dmp.DiffLinesToRunes(e, a)
+	diffs := dmp.DiffMainRunes(fromRunes, toRunes, false)
+	hydrated := make([]diffmatchpatch.Diff, 0, len(diffs))
+	for _, aDiff := range diffs {
+		chars := strings.FieldsFunc(aDiff.Text, func(r rune) bool {
+			return string(r) == diffmatchpatch.IndexSeparator
+		})
+		text := make([]string, len(chars))
+
+		for i, char := range chars {
+			i1, err := strconv.Atoi(char)
+			if err == nil {
+				text[i] = runesToLines[i1]
+			}
+		}
+		for idx, line := range text {
+			if aDiff.Type == diffmatchpatch.DiffEqual && idx < len(text)-1 {
+				continue
+			}
+			hydrated = append(hydrated, diffmatchpatch.Diff{
+				Type: aDiff.Type,
+				Text: line,
+			})
+		}
+	}
+	return hydrated
+}
 
-	diff, _ := difflib.GetUnifiedDiffString(difflib.UnifiedDiff{
-		A:        difflib.SplitLines(e),
-		B:        difflib.SplitLines(a),
-		FromFile: "Expected",
-		FromDate: "",
-		ToFile:   "Actual",
-		ToDate:   "",
-		Context:  1,
-	})
-
-	return "\n\nDiff:\n" + diff
+func prettyDiff(diffs []diffmatchpatch.Diff, useColoredOutput bool) string {
+	var diff strings.Builder
+	if useColoredOutput {
+		diff.WriteString("\\033[31m--- Expected\\033[0m\n\\033[32m+++ Actual\\033[0m\n")
```

Comment:
nit: it may be nice to used constants here for the colors.

Reply:
What's the right place for such constants? Should we keep them in the assert package or move it somewhere else? (e.g. to a `colors` package)